src/clusterfuzz/_internal/bot/tasks/utasks/uworker_io.py

```diff
@@ -99,28 +84,31 @@ def get_entity_with_properties(ndb_key: ndb.Key, properties) -> ndb.Model:
 
 def deserialize_uworker_input(serialized_uworker_input):
   """Deserializes input for the untrusted part of a task."""
-  serialized_uworker_input = json.loads(serialized_uworker_input)
-  uworker_input = serialized_uworker_input['serializable']
-  for name, entity_dict in serialized_uworker_input['entities'].items():
-    entity_key = entity_dict['key']
-    serialized_key = base64.b64decode(bytes(entity_key, 'utf-8'))
-    ndb_key = ndb.Key(serialized=serialized_key)
-    entity = get_entity_with_properties(ndb_key, entity_dict['properties'])
-    uworker_input[name] = UworkerEntityWrapper(entity)
-  return uworker_input
+  uworker_input_proto = uworker_msg_pb2.Input()
+  uworker_input_proto.ParseFromString(serialized_uworker_input)
+  input_dict = {}
+  for descriptor, field in uworker_input_proto.ListFields():
+    if isinstance(field, entity_pb2.Entity):
+      input_dict[descriptor.name] = model._entity_from_protobuf(field)  # pylint: disable=protected-access
+    elif isinstance(field, uworker_msg_pb2.Json):
+      input_dict[descriptor.name] = json.loads(field.serialized)
+    else:
+      input_dict[descriptor.name] = field
+  return input_dict
 
 
 def serialize_uworker_input(uworker_input):
   """Serializes and returns |uworker_input| as JSON. Can handle ndb entities."""
-  serializable = {}
-  ndb_entities = {}
+  uworker_input = uworker_input.copy()
   for key, value in uworker_input.items():
-    if not isinstance(value, ndb.Model):
-      serializable[key] = value
-      continue
-    ndb_entities[key] = make_ndb_entity_input_obj_serializable(value)
+    if isinstance(value, ndb.Model):
+      uworker_input[key] = model._entity_to_protobuf(value)  # pylint: disable=protected-access
```

Comment:
Would it be better to make the function public, given it has been used publicly in different places?